source/common/runtime/runtime_impl.cc

```diff
@@ -244,7 +249,8 @@ void SnapshotImpl::walkDirectory(const std::string& path, const std::string& pre
         entry.uint_value_.value(converted);
       }
 
-      values_[full_prefix] = entry;
+      values_.erase(full_prefix);
```

Comment:
It's unfortunate that this now requires two hash lookups, but I think it's worth it as the interface is much better IMO. I think in C++17 http://en.cppreference.com/w/cpp/container/unordered_map/insert_or_assign could be used here, but I'm not totally sure. I would probably drop a small comment on  why we are doing erase/insert here.

Reply:
thought the same; I'd assume lightweight compared to the IO we're doing before it, though. 

Comment added!